### PR TITLE
Adding manipulathor

### PR DESCRIPTION
**Pull request by ehsanik**

### CODE REVIEW

File: projects/manipulathor_baselines/armpointnav_baselines/models/disjoint_arm_pointnav_models.py
Comment (jordis-ai2) [inline projects/manipulathor_baselines/armpointnav_baselines/models/disjoint_arm_pointnav_models.py:3]: `Arm Point Navigation` ?

File: projects/manipulathor_baselines/armpointnav_baselines/models/arm_pointnav_models.py
Comment (jordis-ai2) [inline projects/manipulathor_baselines/armpointnav_baselines/models/arm_pointnav_models.py:1]: `Arm Point Navigation`?

File: projects/manipulathor_baselines/armpointnav_baselines/models/arm_pointnav_models.py
Comment (jordis-ai2) [inline projects/manipulathor_baselines/armpointnav_baselines/models/arm_pointnav_models.py:3]: (Same as above)

File: projects/manipulathor_baselines/armpointnav_baselines/experiments/ithor/armpointnav_disjoint_depth.py
Comment (jordis-ai2) [inline projects/manipulathor_baselines/armpointnav_baselines/experiments/ithor/armpointnav_disjoint_depth.py:13]: Would it make sense to extend `ArmPointNavDepth`?

**Review by jordis-ai2 — COMMENTED** [review]
First of all, I find it so cool to have this in AllenAct! 👍 

I haven't checked `manipulathor_tasks.py` and `manipulathor_viz.py`, but I believe them to be extensively tested at this level. Other than that, just a couple of minor comments from my side.

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_constants.py:6]: I wasn't sure what this was, can you change it to `MIN_MOVE_THRESHOLD` or similar?

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_constants.py:10]: The `MOVE_ARM_CONSTANT` and `MOVE_ARM_HEIGHT_CONSTANT` constants seem to be `ArmPointNav` specific? Might it make more sense to have these be changeable defaults? See, for instance, how the `VISIBILITY_DISTANCE` constant from the `ithor_plugin` is used.

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_constants.py:12]: I believe the the same comment as above applies to many of the constants below (e.g. if I make a new manipulathor task then I might choose a different set of `TRAIN_OBJECTS`). It might make sense to create a `armpointnav_constants` file and add them there instead.

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_constants.py:86]: These seem like utilities?

File: allenact_plugins/manipulathor_plugin/manipulathor_environment.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_environment.py:206]: Why explicitly remove this? I'd also change this exception to a `NotImplementedError`.

File: allenact_plugins/manipulathor_plugin/manipulathor_sensors.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_sensors.py:50]:
```
return np.zeros_like(env.current_frame)
```
is a bit shorter.

File: allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py:133]: Shouldn't this exception be fatal?

Comment (Lucaweihs) [reply allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py:133]: Some other parts of the codebase have this problem too but we also need to find a better way to handle dataset paths that might not exist. Maybe add a `TODO` saying that this should be cleaned up.

File: allenact_plugins/manipulathor_plugin/manipulathor_environment.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/manipulathor_environment.py:38]: This spacing suggests to me that black hasn't been run on this file before committing. Can you call `bash scripts/auto_format.sh` from top level directory of the repository?

**Review by Lucaweihs — COMMENTED** [review]
I haven't had the chance to read everything in detail but thought it was better to give some comments now rather than have you wait longer. Obviously this is an awesome addition to allenact. Other than some small style/consistency suggestions, my general suggestion is to (whenever possible) separate the pieces of the code that are `ArmPointNav`-specific and those that are more generally applicable (e.g. there are some constants in `manipulathor_constants.py` that I could see myself wanting to change if I was doing a different task).

File: allenact_plugins/manipulathor_plugin/arm_calulcation_util.py
Comment (jordis-ai2) [reply allenact_plugins/manipulathor_plugin/arm_calulcation_util.py:44]: Yes, I had missed there was the translation component 👍

File: allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py
Comment (ehsanik) [reply allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py:133]: Well, this is in case an object does not exist in a room. It's not fatal. Just the dataset won't include that object in that specific room. I can log errors if you think that makes more sense.

File: allenact_plugins/manipulathor_plugin/manipulathor_environment.py
Comment (ehsanik) [reply allenact_plugins/manipulathor_plugin/manipulathor_environment.py:177]: randomizing agent is done differently than the other frameworks, this is due to the collision of the arm during agent teleport which needs to be taken care of. That's why I prefer to make sure users don't use this function to get agent locations.

File: allenact_plugins/manipulathor_plugin/manipulathor_environment.py
Comment (ehsanik) [reply allenact_plugins/manipulathor_plugin/manipulathor_environment.py:38]: Done

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (ehsanik) [reply allenact_plugins/manipulathor_plugin/manipulathor_constants.py:16]: great idea. I did that and added it under manipulathor_plugin. But it kinda feels like to be in the wrong place. Should I move it to projects/manipulathor_baselines/armpointnav_baselines? What do you think?

File: tests/manipulathor_plugin/test_utils.py
Comment (jordis-ai2) [reply tests/manipulathor_plugin/test_utils.py:19]: ... and use `abs()` for each difference?

**Review by jordis-ai2 — COMMENTED** [review]
Just a minor comment. Other than that, it LGTM 👍

File: tests/manipulathor_plugin/test_utils.py
Comment (ehsanik) [reply tests/manipulathor_plugin/test_utils.py]: Nice catch Jordi :) Fixed it

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (Lucaweihs) [reply allenact_plugins/manipulathor_plugin/manipulathor_constants.py:9]: I like that idea :)!

File: allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py
Comment (Lucaweihs) [reply allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py:132]: I think logging errors would make sense to me, maybe just replacing this with `get_logger().error(f"Failed to load {valid_position_adr}")` would be good.

File: allenact_plugins/manipulathor_plugin/armpointnav_constants.py
Comment (Lucaweihs) [inline allenact_plugins/manipulathor_plugin/armpointnav_constants.py:1]: I'm happy with the constants being moved here too :)

File: allenact_plugins/manipulathor_plugin/manipulathor_constants.py
Comment (Lucaweihs) [reply allenact_plugins/manipulathor_plugin/manipulathor_constants.py:9]: Didn't realize you'd already moved these, they look good at `allenact_plugins/manipulathor_plugin/armpointnav_constants.py`

**Review by Lucaweihs — APPROVED** [review]
There seem to be a few last LGTM notes, after you fix those feel free to merge from my perspective!

**Pull request merged**
